aws-java-sdk-core/src/main/java/com/amazonaws/auth/profile/internal/securitytoken/ProfileCredentialsService.java

```diff
@@ -0,0 +1,10 @@
+package com.amazonaws.auth.profile.internal.securitytoken;
+
+import com.amazonaws.auth.AWSCredentials;
+import com.amazonaws.auth.AWSCredentialsProvider;
+
+public interface ProfileCredentialsService {
+    AWSCredentialsProvider assumeRole(AWSCredentials longLivedCredentials, RoleInfo targetRoleInfo);
```

Comment:
What do you think about moving the long lived credentials into RoleInfo and having a convenience setter for AWSCredentials that wraps it in a static provider. Could simplify the interface a bit.